Library/Formula/libstfl.rb

```diff
@@ -18,6 +18,10 @@ class Libstfl < Formula
   patch :DATA
 
   def install
+    if Formula["ruby"].installed?
```

Comment:
@DomT4 Hehe, you certainly aren't an idiot, but you do happen to be wrong.

If you add <code>depends_on :ruby => ["1.8", :build]</code>, the path changes from
```
/usr/local/Library/ENV/4.3:/usr/local/opt/swig/bin:/usr/bin:/bin:/usr/sbin:/sbin
```
to
```
/usr/local/Library/ENV/4.3:/usr/local/opt/swig/bin:/usr/local/opt/ruby/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

The RubyRequirement checks Version.new(version.to_s) >= Version.new(@version), and then assuming the dependency is satisfied, does `paths += deps.map { |d| d.opt_bin.to_s }` here:
https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/extend/ENV/super.rb#L117
which changes the PATH from
`[#<Pathname:/usr/local/Library/ENV/4.3>]`
to
`[#<Pathname:/usr/local/Library/ENV/4.3>, "/usr/local/opt/swig/bin", "/usr/local/opt/ruby/bin"]`